spicy/toolchain/src/compiler/validator.cc

```diff
@@ -799,14 +799,17 @@ struct VisitorPost : visitor::PreOrder, hilti::validator::VisitorMixIn {
         }
 
         if ( count_attr && (repeat && ! repeat->type()->type()->isA<hilti::type::Null>()) )
-            error("cannot have both `[..]` and &count", n);
+            error("cannot have both '[..]' and &count", n);
+
+        if ( count_attr )
+            deprecated("&count is deprecated, prefer '[..]' syntax", count_attr->meta().location());
```

Comment:
How about for slightly more clarity:

```suggestion
            deprecated("&count=N is deprecated, prefer '[N]' syntax", count_attr->meta().location());
```